instrumentation-genai/opentelemetry-instrumentation-vertexai/src/opentelemetry/instrumentation/vertexai/utils.py

```diff
@@ -137,3 +145,46 @@ def get_span_name(span_attributes: Mapping[str, AttributeValue]) -> str:
     if not model:
         return f"{name}"
     return f"{name} {model}"
+
+
+def request_to_events(
+    *, params: GenerateContentParams, capture_content: bool
+) -> Iterable[Event]:
+    # System message
+    if params.system_instruction:
```

Comment:
does vertex support tool calling? is it something you're considering as a follow up?

Reply:
It does, tracking in https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3216